src/main/java/com/faforever/client/FafClientApplication.java

```diff
@@ -78,6 +80,31 @@ public void init() throws InterruptedException, NoSuchFieldException, IllegalAcc
         .run(getParameters().getRaw().toArray(new String[0]));
 
     OperatingSystem operatingSystem = applicationContext.getBean(OperatingSystem.class);
+
+    try {
+      String preferencesDirectoryString = String.valueOf(operatingSystem.getPreferencesDirectory());
+      log.debug("Current preferences directory " + preferencesDirectoryString);
+      if (preferencesDirectoryString.matches(".*[а-яА-ЯёЁ].*")) {
+        I18n i18n = applicationContext.getBean(I18n.class);
+        String warningMessage = i18n.get("home.directory.warning.cyrillic");
+        String closeText = i18n.get("close");
+        ButtonType closeButton = new ButtonType(closeText, ButtonData.CANCEL_CLOSE);
+        CountDownLatch waitForUserInput = new CountDownLatch(1);
+
+        JavaFxUtil.runLater(() -> {
+          Alert alert = new Alert(AlertType.WARNING, warningMessage, closeButton);
+          Optional<ButtonType> buttonType = alert.showAndWait();
+          if (buttonType.filter(button -> button == closeButton).isPresent()) {
+            System.exit(EXIT_STATUS_CYRILLIC_CHARACTER_PREFERENCES_DIRECTORY);
+          }
+          waitForUserInput.countDown();
+        });
+        waitForUserInput.await();
+      }
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
```

Comment:
The try catch is unnecessary as the method declares that it throws InterruptedException